### PR TITLE
Update to node 16 and npm 8 via `nvm` and updated many node packages

### PR DESCRIPTION
**Pull request by AndrewGable**
### Details

Update to Node 16 and npm 8, there were quite a few updates that needed to be applied due to the peer dependency  changes in npm 7. In npm 7 peer dependencies are installed by default, which creates a much stricter way of installing peer dependencies. For example, `react-native-web` upgraded to React 18, so now all other packages needed to be upgraded to use React 18.

This caused us to need to upgrade all of these packages:
* `@onfido/react-native-sdk` - React 18
* `@pieter-pot/react-native-fast-image` - React 18
* `@react-native-community/cameraroll` - React 18
* `react-dom` - React 18
* `react-pdf` - React 18
* `react-plaid-link` - React 18
* `@storybook/addon-react-native-web` - React 18
* `@welldone-software/why-did-you-render` - React 18
* `babel-plugin-react-native-web` - `react-native-web` upgrade
* `eslint-plugin-jsx-a11y` - Wasn't installed previously
* `react-native-flipper` - React 18/ RN upgrade
* `rn-async-storage-flipper` - Removed, I verified with @marcaaron this was OK.
* `tests/unit/nativeVersionUpdaterTest.js` - Changes to Node 16 updated function return type

### Fixed Issues

$ https://github.com/Expensify/App/issues/10331

### Tests

- [x] Verify that no errors appear in the JS console
1. Run a general "smoke" test of the app by visiting lots of pages and doing lots of actions.

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### QA Steps

1. Run a smoke test of the app, lots of libraries were changed

#### Web

Screen Shot 2022-08-18 at 2 57 50 PM

#### Desktop

Screen Shot 2022-08-18 at 3 00 38 PM

#### iOS

Screen Shot 2022-08-18 at 2 56 35 PM

Screen Shot 2022-08-18 at 2 56 31 PM

#### Android
Screen Shot 2022-08-18 at 2 44 50 PM
Screen Shot 2022-08-18 at 2 44 58 PM

### CODE REVIEW

File: package.json
Comment (AndrewGable) [inline package.json:46]: Hack: We should fork this under Expensify org.

File: package.json
Comment (AndrewGable) [inline package.json:89]: Hack: This is a contributors fork, but this repo looks unmaintained.

File: package.json
Comment (AndrewGable) [inline package.json:132]: Hack: This is a PR, but PR has been open for quite some time.

File: package.json
Comment (AndrewGable) [inline package.json:173]: Hack: This was merged a few hours before I made this PR, I assume it will land soon-ish.

File: .github/workflows/platformDeploy.yml
Comment (AndrewGable) [inline .github/workflows/platformDeploy.yml:34]: @roryabraham - I was unsure what to do here, if we leave as `main`, it will use an older version of `node` and fail. So I think this is necessary until we merge to `main`, then change it back.

Comment (roryabraham) [reply .github/workflows/platformDeploy.yml:34]: Another alternative would be to just checkout and use the local version of the action (no @ tag). But honestly I think this option is fine – we can just do a fast-follower PR to switch it back to main

Comment (AndrewGable) [reply .github/workflows/platformDeploy.yml:34]: Sounds good 👍

**Comment by AndrewGable** [comment]
GitHub actions are finally happy. I will work on cleaning this up some more, then open it up for review once it's clean!

File: package.json
Comment (AndrewGable) [reply package.json]: Updated to a published fork.

File: package.json
Comment (AndrewGable) [reply package.json:46]: Done here: https://github.com/Expensify/react-native-sdk/pull/1 - Just waiting for it to be merged.

Comment (AndrewGable) [reply package.json:46]: Fixed this!

File: package.json
Comment (AndrewGable) [reply package.json:174]: According to their GitHub, this should land in `0.160.0` but I haven't seen it hit the npm registry yet.

**Review by luacmartins — APPROVED** [review]
LGTM and tests well on all platforms!

**Review by NikkiWines — APPROVED** [review]
👍 aside from Rajat's comment!

File: .github/workflows/platformDeploy.yml
Comment (AndrewGable) [reply .github/workflows/platformDeploy.yml:34]: I think the idea is that we want to lock it to a branch, but we cannot lock it to `main` until this PR is merged. Using local files would work, but it wouldn't lock it to a commit.

**Review by parasharrajat — APPROVED** [review]

File: tests/unit/nativeVersionUpdaterTest.js
Comment (roryabraham) [inline tests/unit/nativeVersionUpdaterTest.js:81]: NAB because I don't think we've standardized on this, but generally `.then` goes on the next line from what I've seen

File: src/setup/platformSetup/index.native.js
Comment (roryabraham) [inline src/setup/platformSetup/index.native.js:28]: Why is this not needed anymore?

**Review by roryabraham — APPROVED** [review]
No blockers, LGTM 👍

File: ios/Podfile
Comment (marcaaron) [inline ios/Podfile:10]: Is this auto-linked in a newer version?

File: src/setup/platformSetup/index.native.js
Comment (AndrewGable) [reply src/setup/platformSetup/index.native.js:28]: 1. It was breaking due to peer dependencies and the library not being maintained
2. I believe @marcaaron said this was a debugging tool that is not used anymore

**Review by marcaaron — APPROVED** [review]
npm install worked well for me after adding the correct node version

File: ios/Podfile
Comment (AndrewGable) [reply ios/Podfile:10]: Yes correct. I don't know the history behind why we added this, but I think we should rely on auto linking all pod libraries unless there is a specific version breaking for us.

File: src/setup/platformSetup/index.native.js
Comment (marcaaron) [reply src/setup/platformSetup/index.native.js:28]: It might be used by some (but I haven't used it in a long time). There are workarounds to this package e.g. if you need to know the contents of AsyncStorage you can always log out values. Might be interesting to do our own Flipper plugin for Onyx at some point 🤔

File: ios/Podfile
Comment (AndrewGable) [reply ios/Podfile:10]: I don't understand their documentation.. It says it uses auto linking, so we shouldn't need to add it to the Podfile manually. 

Plaid 2.5.1 is installed via autolinking and can be seen in the Podfile.lock: https://github.com/Expensify/App/pull/10439/files#diff-b7d85c35701918f25c9b2c418249f9b66fd9b067676e295a9eff493b73ee1752R231

And the Plaid SDK seems to boot just fine:
Screen Shot 2022-08-19 at 12 30 39 PM
Screen Shot 2022-08-19 at 12 31 34 PM

**Comment by AndrewGable** [comment]
Ok all reviewers have approved so I'm going to:

File: package.json
Comment (mdneyazahmad) [inline package.json:190]: This produces error when the node and npm version does not match exactly. Developer has to use the exact version. Maybe, we are fine with `16.x` and `8.x`.

Comment (luacmartins) [reply package.json:190]: I think the error is intentional and we want to use the exact version to avoid any harder to debug errors. It should be easy to install and change the version of node if you use nvm.

**Pull request merged**
